utils/png_info.py

```
@@ -0,0 +1,252 @@
+            # The image mode (RGB, RGBA, etc.) doesn't affect text chunk extraction
+            # Text chunks are stored separately from image data in PNG files
+            
+            if hasattr(img, 'text') and img.text:
+                # Return the text chunks as a dictionary
+                # This works for all image modes: RGB, RGBA, L, LA, P, etc.
+                return dict(img.text)
+            
+            return None
+            
+    except Exception as e:
+        raise IOError(f"Failed to parse PNG metadata: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)